File: Java/DesignPatterns/src/main/java/designPatterns/behavioral/visitor/DisplayVisitor.java
package designPatterns.behavioral.visitor;

public class DisplayVisitor implements Visitor {
  @Override
  public String visit(ConcretePart.A a) {
    return "Displaying A";
  }

  @Override
  public String visit(ConcretePart.B b) {
    return "Displaying B";
  }

  @Override
  public String visit(ConcretePart.C c) {
    return "Displaying C";
  }
}

File: Kotlin/DesignPatterns/src/main/kotlin/behavioral/visitor/DisplayVisitor.kt
package behavioral.visitor

class DisplayVisitor : Visitor {
  override fun visit(a: ConcretePart.A): String = "Displaying A"
  override fun visit(b: ConcretePart.B): String = "Displaying B"
  override fun visit(c: ConcretePart.C): String = "Displaying C"
}